jba/src/processing/tasktracker_content_collector.py

```diff
@@ -0,0 +1,128 @@
+import argparse
+import itertools
+from dataclasses import dataclass
+from pathlib import Path
+from typing import List, Dict
+
+from collect_course_structure import gather_structure
+from core.src.utils.file.file_utils import find_files_by_regex
+from core.src.utils.file.yaml_utils import read_yaml_field_content, save_as_yaml
+from jba.src.models.edu_structure import (
+    EduStructureType,
+    EduStructureNode,
+    EduInfoFileField,
+    INFO_FILE_REGEX,
+)
+
+CONTENT_FILE_NAME = 'task_content_default.yaml'
+FRAMEWORK_TYPE = 'framework'
+TASK_DIRECTORY_NAME = 'task'
+EXTENSIONS = {'py': 'PYTHON', 'ipynb': 'JUPYTER', 'java': 'JAVA', 'kt': 'KOTLIN', 'cpp': 'CPP', 'csv': 'CSV'}
+
+
+@dataclass(eq=True, frozen=True)
+class TaskTrackerFile:
+    rel_path: Path
+
+    @property
+    def name(self) -> str:
+        return self.rel_path.stem
+
+    @property
+    def path(self) -> Path:
+        return self.rel_path.parent
+
+    @property
+    def extension(self) -> str:
+        return EXTENSIONS.get(self.rel_path.suffix.lstrip('.'), 'NO_EXTENSION')
+
+    def as_dict(self) -> dict:
+        return {
+            'filename': self.name,
+            'relativePath': str(self.path),
+            'extension': self.extension,
+            'sourceSet': 'SRC',
+            'isInternal': False
+        }
+
+
+def get_data_template(files: List[Dict]) -> Dict:
+    return {
+        'tasks': [{
+            'name': 'example',
+            'description': 'description',
+            'id': 'main',
+            'files': files
+        }]
+    }
+
+
+def get_info_file(root: Path) -> Path:
+    info_files = find_files_by_regex(root, INFO_FILE_REGEX)
+    return info_files[0]
+
+
+def get_files(course_root: Path, relative_path: Path, lesson: EduStructureNode) -> List[TaskTrackerFile]:
+    lesson_path = course_root / relative_path
+    info_file = get_info_file(lesson_path)
+    yaml_file_content = read_yaml_field_content(lesson_path / info_file, EduInfoFileField.TYPE.value)
+    return itertools.chain.from_iterable(
+        [get_task_files(lesson_path / child.name,
+                        relative_path,
+                        yaml_file_content is not None and yaml_file_content == FRAMEWORK_TYPE)
+         for child in lesson.children])
+
+
+def get_task_files(root: Path, relative_path: Path, is_framework: bool) -> List[TaskTrackerFile]:
```

Comment:
I suggest simply renaming 'root' to 'task_root.' I believe there is no point in passing the project root since it is not used anywhere. What we need is the path to the task file and the relative path from the root to the files.